urllib3/contrib/pyopenssl.py

```diff
@@ -204,9 +204,11 @@ def _send_until_done(self, data):
                 continue
 
     def sendall(self, data):
-        while len(data):
-            sent = self._send_until_done(data)
-            data = data[sent:]
+        total_sent = 0
+        while total_sent < len(data):
+            # OpenSSL will only write 16K at once
+            sent = self._send_until_done(data[total_sent:total_sent+16384])
```

Comment:
Would have been nice for 16384 to not be a magic number, e.g., make it a constant that's named so we ave context for the future. (The comment alleviates this, but it's nice to have the comment and the constant in my opinion. And it's irrelevant if @shazow disagrees ;).)